### ISSUE TITLE
ion-datetime translation issue

### ISSUE THREAD

**Issue by AndrewKuzich** [ionic-team/ionic#11034]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[x] **2.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
There is a problem with ion-datetime translation. After monthNames is set it cannot be changed without page reload | rerender. I use ngx-translate https://github.com/ngx-translate/core module in my project for i18n issues. User can change current language.
e.g. default lang is en, ion-datetime is rendered with english months. Then user changes current language to german. monthNames variable is also changed. BUT month names in the date-picker modal are not changed. NOTICE: there is no page reload after language is changed

**Expected behavior:**
monthNames should be changed in picker without page reload

**Comment by AndrewwKuzich** [comment]
After page is reloaded (destroyed and created) correct month translations are applied

**Comment by jgw96** [comment]
Hello, thanks for opening an issue with us. We will look into this.

**Comment by digaus** [comment]
Can confirm... same thing with day names and shorts.
